crates/topos-sequencer-tce-proxy/src/lib.rs

```diff
@@ -491,11 +492,11 @@ impl TceProxyWorker {
                     // process TCE proxy commands received from application
                     Some(cmd) = command_rcv.recv() => {
                         match cmd {
-                            TceProxyCommand::SubmitCertificate(cert) => {
-                                info!(
-                                    "Submitting new certificate to the TCE network: {:?}",
-                                    &cert
-                                );
+                            TceProxyCommand::SubmitCertificate{cert, ctx} => {
+                                let span = info_span!("Sequencer TCE Proxy");
+                                span.set_parent(ctx);
+                                let _span = span.enter();
```

Comment:
You should not use `enter` in an async method because it will produce incorrect traces.
You should use `entered` method:
```suggestion
                                let _span = span.entered();
```

or even `in_scope`:
```suggestion
                                span.in_scope(|| { .. });
```